File: cubby/FractionFormatter/src/org/acm/mbaechi/fractions/AbstractFractionRounder.java
package org.acm.mbaechi.fractions;

public abstract class AbstractFractionRounder<T> implements FractionRounder<T> {
protected final double divisor;
	
	AbstractFractionRounder(int fraction) {
		this.divisor = 1d / fraction;		
	}
}

File: cubby/csharp/FractionFormatter/FractionFormatter/AbstractFractionRounder.cs
using System;

namespace FractionFormatter
{
	public abstract class AbstractFractionRounder<T> : FractionRounder<T> {

	    protected readonly double divisor;

	    protected AbstractFractionRounder(int fraction) {
	        this.divisor = 1d / fraction;
	    }

		public abstract T round(T number);
	}
}

